flake.nix

```diff
@@ -1,48 +1,74 @@
 {
   inputs = {
-    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2305.*.tar.gz";
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2405.*";
 
-
-    rust-overlay = {
-      url = "github:oxalica/rust-overlay";
+    fenix = {
+      url = "https://flakehub.com/f/nix-community/fenix/0.1.*";
       inputs.nixpkgs.follows = "nixpkgs";
-      inputs.flake-utils.follows = "flake-utils";
     };
 
     crane = {
-      url = "https://flakehub.com/f/ipetkov/crane/0.14.*.tar.gz";
+      url = "https://flakehub.com/f/ipetkov/crane/0.14.*";
       inputs.nixpkgs.follows = "nixpkgs";
       inputs.flake-compat.follows = "flake-compat";
-      inputs.flake-utils.follows = "flake-utils";
     };
 
-    flake-compat.url = "https://flakehub.com/f/edolstra/flake-compat/1.0.1.tar.gz";
+    flake-compat.url = "https://flakehub.com/f/edolstra/flake-compat/*";
+    flake-schemas.url = "https://flakehub.com/f/DeterminateSystems/flake-schemas/*";
   };
 
-  outputs = { self, nixpkgs, flake-utils, rust-overlay, crane, ... }:
+  outputs = { self, nixpkgs, fenix, crane, flake-schemas, ... }:
     let
       supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
       forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f rec {
         pkgs = import nixpkgs {
           inherit system;
-          overlays = [
-            rust-overlay.overlays.default
-          ];
-        };
-
-        cranePkgs = pkgs.callPackage ./crane.nix {
-          inherit crane supportedSystems;
-          darwinFrameworks = with pkgs.darwin.apple_sdk.frameworks; [ Security SystemConfiguration ];
+          overlays = [ self.overlays.default ];
         };
       });
+      meta = (builtins.fromTOML (builtins.readFile ./Cargo.toml)).package;
     in
     {
-      packages = forAllSystems ({ cranePkgs, ... }: rec {
-        inherit (cranePkgs) flake-checker;
+      inherit (flake-schemas) schemas;
+
+      overlays.default = final: prev: rec {
+        system = final.stdenv.hostPlatform.system;
+        rustToolchain = with fenix.packages.${system};
+          combine ([
+            stable.clippy
+            stable.rustc
+            stable.cargo
+            stable.rustfmt
+            stable.rust-src
+            stable.rust-analyzer-preview
+          ] ++ nixpkgs.lib.optionals (system == "x86_64-linux") [
+            targets.x86_64-unknown-linux-musl.stable.rust-std
+          ] ++ nixpkgs.lib.optionals (system == "aarch64-linux") [
+            targets.aarch64-unknown-linux-musl.stable.rust-std
+          ]);
+        craneLib = (crane.mkLib final).overrideToolchain rustToolchain;
+        darwinInputs = final.lib.optionals final.stdenv.isDarwin (with final.darwin.apple_sdk.frameworks; [ Security SystemConfiguration ]);
+      };
+
+      packages = forAllSystems ({ pkgs }: rec {
```

Comment:
Might be a good idea to test that this still works on both Linux and macOS -- I remember we had an issue with flakehub-push in the past not being as static as we expected it to be last time we made a similar change to its flake.nix.

Reply:
Okay, I added a CI test that just `nix run`s it on both macOS and Linux. I'm not sure if that's sufficient so plz advise if not.

Reply:
I meant more an "imperative" test (i.e. upload this to S3 and then use it in an actual action in some other repository, like we require for update-flake-lock, although that doesn't have a binary to worry about). I don't think that this would be quite enough to catch the "not static" case, as it'll still have been built on the same machine.

Reply:
Ah, we actually _do_ have a test for that that's currently failing 😄 I'll investigate.

Reply:
Okay, the static Nix build should be fixed now (reverted some changes but updated the inputs to get a more recent Rust)